driver/resource_pool.go

```diff
@@ -19,15 +20,44 @@ func (d *Driver) NewResourcePool(ref *types.ManagedObjectReference) *ResourcePoo
 	}
 }
 
-func (d *Driver) FindResourcePool(host string, name string) (*ResourcePool, error) {
-	p, err := d.finder.ResourcePool(d.ctx, fmt.Sprintf("/%v/host/%v/Resources/%v", d.datacenter.Name(), host, name))
-	if err != nil {
-		return nil, err
+func (d *Driver) FindResourcePool(name string) (*ResourcePool, error) {
+	t := d.client.ServiceContent.About.ApiType
+	switch t {
+	case "HostAgent":
+		ddc, err := d.finder.DefaultDatacenter(d.ctx)
+		if err != nil {
+			return nil, err
+		}
+		d.finder.SetDatacenter(ddc)
+		p, err := d.finder.DefaultResourcePool(d.ctx)
+		if err != nil {
+			return nil, err
+		}
+		return &ResourcePool{
+			pool:   p,
+			driver: d,
+		}, nil
+	case "VirtualCenter":
+		if name != "" {
+			p, err := d.finder.ResourcePool(d.ctx, name)
```

Comment:
this changes a syntax. previously pools name were relative to a host. Now a full path is required: 
```
esxi-1.vsphere65.test/Resources
esxi-1.vsphere65.test/Resources/pool1
cluster1/Resources
cluster2/Resources
cluster2/Resources/pool1
```

Reply:
I can't try this by myself (my cluster has no DSR and I can't create `resource_pools`) but checking the [code](https://github.com/vmware/govmomi/blob/master/find/finder.go#L845) I think that you can either provide only the name of the `resource_pool`or the complete path.

In the examples you provide you could use only the name of the pool, considering there exist a unique `resource_pool`with that name in your vCenter, otherwise, specifying only the name `pool1`, you would get the error:

```
==> vsphere-iso: Creating VM...
Build 'vsphere-iso' errored: error creating vm: default resource pool resolves to multiple instances, please specify
```

In this case, you would have to use the complete path of the `resource_pool`, i.e. `cluster2/Resources/pool1`

This same thing is happening when trying to find de default `resource_pool`. Checking the [code](https://github.com/vmware/govmomi/blob/master/find/finder.go#L858), we are searching for  `*/Resources`, the root `resurce_pool`. In your example, `DefaultResourcePool`would fail again with the previous error, because we have 3 different root `resource_pools`on the vCenter (`esxi-1.vsphere65.test/Resources`, `cluster1/Resources` and `cluster2/Resources`) . Again, in this case we would have to use the complete path of the `resource_pool`.

I don't see this a problem, as long as your pool name is unique, you can use only the name; if your vCenter has only one root `resource_pool`, you can leave it blank, and in other cases (repeated pool name or more than one root `resource_pool`), you will have to specify the complete path.